File: commands/link.js
import { EmbedBuilder } from 'discord.js';
import links from '../models/links';
import db from '../services/db';
import colors from '../models/colors';

export const helpers = {
	ephemeral: false,
};

export async function run(interaction) {
	const social = interaction.options.getString('сайт');
	const socials = await db.one('SELECT * FROM userUrls WHERE id = ?', [interaction.guildId]);
	console.log(socials);

	if (!socials?.[social !== 'custom' ? social : 'website']) {
		await interaction.deleteReply();
		await interaction.followUp({ content: 'Такая ссылка у сервера не задана!', ephemeral: true });
		return;
	}

	const link = links[social];
	const url = link.url + socials?.[social !== 'custom' ? social : 'website'] ?? '';
	const embed = new EmbedBuilder()
		.setColor(colors.blue)
		.setFooter({ text: 'Ссылка, указанная на сайте мониторинга.' })
		.setAuthor({
			name: link.name,
			iconURL: link.icon?.length > 0 ? link.icon : null,
			url: url?.length > 0 ? url : null,
		});

	await interaction.editReply({ embeds: [embed] });
}

export default {
	helpers,
	run,
};

File: commands/link.go
package commands

import (
	"context"

	"github.com/bwmarrin/discordgo"

	"sdc/domain"
	"sdc/models"
)

func Link(d *Deps, s *discordgo.Session, i *discordgo.InteractionCreate) error {
	ctx := context.Background()
	social, _ := optionString(i, "сайт")
	linkMeta, ok := domain.SocialLinks[social]
	if !ok {
		deleteDeferred(s, i)
		return followupEphemeral(s, i, &discordgo.WebhookParams{Content: "Неизвестный тип ссылки!"})
	}

	socials, err := d.Store.GuildSocials(ctx, i.GuildID)
	if err != nil {
		return err
	}
	key := domain.SocialKey(social)
	path := ""
	if socials != nil {
		path = socials[key]
	}
	if path == "" {
		deleteDeferred(s, i)
		return followupEphemeral(s, i, &discordgo.WebhookParams{Content: "Такая ссылка у сервера не задана!"})
	}

	url := linkMeta.URL + path
	author := &discordgo.MessageEmbedAuthor{Name: linkMeta.Name, URL: url}
	if linkMeta.Icon != "" {
		author.IconURL = linkMeta.Icon
	}

	embed := &discordgo.MessageEmbed{
		Color:  models.EmbedColors["blue"],
		Author: author,
		Footer: &discordgo.MessageEmbedFooter{Text: "Ссылка, указанная на сайте мониторинга."},
	}
	return editReply(s, i, &discordgo.WebhookEdit{Embeds: &[]*discordgo.MessageEmbed{embed}})
}
